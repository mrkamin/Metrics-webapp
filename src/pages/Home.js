import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import StockMaterialsLists from '../components/MatrialList';
import { filterCompany } from '../components';
import style from '../style/Home.module.css';

const Home = () => {
  const navigate = useNavigate();
  const stockMaterials = useSelector((store) => store.stockMaterials);
  const [stockMaterial, setStockMaterial] = useState([]);

  useEffect(() => {
    setStockMaterial(stockMaterials);
  }, [stockMaterials]);

  const handleSelection = (company) => {
    const filtered = filterCompany(company, stockMaterials);
    setStockMaterial(filtered);
  };

  const handleSearch = (text) => {
    // eslint-disable-next-line
    const filteredCompanies = stockMaterials.filter((ex) => ex.companyName.toLowerCase().includes(text));
    setStockMaterial(filteredCompanies);
  };

  const showStockDetails = (id) => {
    const item = stockMaterials.filter((material) => material.symbol === id);
    navigate('/details', { state: item[0] });
  };

  return (
    <div className={style.homeContainer}>
      <div className={style.homeHederBackground}>
        <div className={style.homeHeder}>
          <h1 className={style.homeTitle}>Companies</h1>
          <input className={style.homeInput} type="text" onChange={(e) => handleSearch(e.target.value)} placeholder="search companies" />
          <form>
            <select className={style.homeform} id="companies" name="companies" onChange={(e) => handleSelection(e.target.value)}>
              <option value="Select A Company">Select A Company</option>
              <option value="allCompanies">All Companies</option>
              <option value="AppleInc">Apple Inc</option>
              <option value="NvidiaCorporation">Nvidia Corporation</option>
              <option value="BroadcomInc">Broadcom Inc</option>
              <option value="ASMLHolding">ASML Holding</option>
              <option value="AdobeInc">Adobe Inc</option>
              <option value="TexasInstrumentsIncorporated">Texas Instruments Incorporated</option>
              <option value="Advanced_Micro_Devices_Inc">Advanced Micro Devices Inc</option>
              <option value="QualcommIncorporated">Qualcomm Incorporated</option>
              <option value="IntuitInc">Intuit Inc</option>
              <option value="AnalogDeviceInc">Analog Device Inc</option>
              <option value="LamResearchCorppration">Lam Research Corppration</option>
              <option value="MicronTechnologyINC">Micron_Technology_INC</option>
              <option value="PaloAltoNetworksINC">Palo_Alto_Networks_INC</option>
              <option value="SynopsysInc">Synopsys_Inc</option>
              <option value="otherCompanies">Other Companies</option>
            </select>
          </form>
        </div>
      </div>
      <div className={style.materialListContainer}>
        {stockMaterial.map((stockScreener) => (
          <StockMaterialsLists
            key={stockScreener.symbol}
            stockScreener={stockScreener}
            showStockDetails={showStockDetails}
          />
        ))}
      </div>
      <div>
        <span className={style.homeFooter}>Stock Screener Copywrite Mohammad Rafi Amin</span>
      </div>
    </div>
  );
};

export default Home;
