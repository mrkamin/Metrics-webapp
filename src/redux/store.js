import { configureStore } from '@reduxjs/toolkit';
import stockReducer from './Stock-Screener/stockScreener';

const store = configureStore({
  reducer: {
    stockMaterials: stockReducer,
  },
});

export default store;
