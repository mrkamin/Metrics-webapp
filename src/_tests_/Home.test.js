import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Home from '../pages/Home';
import renderProvider from '../redux/Stock-Screener/stockScreener-test';

it('snapshots', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('testing the heading inside home component', async () => {
  renderProvider(
    <Router>
      <Home />
    </Router>,
  );
  expect(screen.getByText(/Select A Company/i)).toBeInTheDocument();
});
