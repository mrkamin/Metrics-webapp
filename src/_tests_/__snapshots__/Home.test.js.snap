// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="homeContainer"
>
  <div
    className="homeHederBackground"
  >
    <div
      className="homeHeder"
    >
      <h1
        className="homeTitle"
      >
        Companies
      </h1>
      <input
        className="homeInput"
        onChange={[Function]}
        placeholder="search companies"
        type="text"
      />
      <form>
        <select
          className="homeform"
          id="companies"
          name="companies"
          onChange={[Function]}
        >
          <option
            value="Select A Company"
          >
            Select A Company
          </option>
          <option
            value="allCompanies"
          >
            All Companies
          </option>
          <option
            value="Apple_Inc"
          >
            Apple Inc
          </option>
          <option
            value="Nvidia_Corporation"
          >
            Nvidia Corporation
          </option>
          <option
            value="Broadcom_Inc"
          >
            Broadcom Inc
          </option>
          <option
            value="ASML_Holding"
          >
            ASML Holding
          </option>
          <option
            value="Adobe_Inc"
          >
            Adobe Inc
          </option>
          <option
            value="Texas_Instruments_Incorporated"
          >
            Texas Instruments Incorporated
          </option>
          <option
            value="Advanced_Micro_Devices_Inc"
          >
            Advanced Micro Devices Inc
          </option>
          <option
            value="Qualcomm_Incorporated"
          >
            Qualcomm Incorporated
          </option>
          <option
            value="Intuit_Inc"
          >
            Intuit Inc
          </option>
          <option
            value="Analog_Device_Inc"
          >
            Analog Device Inc
          </option>
          <option
            value="Lam_Research_Corppration"
          >
            Lam Research Corppration
          </option>
          <option
            value="Micron_Technology_INC"
          >
            Micron_Technology_INC
          </option>
          <option
            value="Palo_Alto_Networks_INC"
          >
            Palo_Alto_Networks_INC
          </option>
          <option
            value="Synopsys_Inc"
          >
            Synopsys_Inc
          </option>
          <option
            value="otherCompanies"
          >
            Other Companies
          </option>
        </select>
      </form>
    </div>
  </div>
  <div
    className="materialListContainer"
  />
  <div>
    <span
      className="homeFooter"
    >
      Stock Screener Copywrite Mohammad Rafi Amin
    </span>
  </div>
</div>
`;
